EIPS/eip-6404.md

```diff
@@ -0,0 +1,1000 @@
+| `TX_HASH_VERSION_EIP2930` | `TxHashVersion(0x02)` | [EIP-2930](./eip-2930.md#definitions) transaction |
+| `TX_HASH_VERSION_EIP1559` | `TxHashVersion(0x03)` | [EIP-1559](./eip-1559.md#specification) transaction |
+| `TX_HASH_VERSION_EIP4844` | `TxHashVersion(0x04)` | [EIP-4844](./eip-4844.md#parameters) transaction |
+
+#### Execution block changes (Transactions)
+
+A new `Transaction` SSZ container is introduced to represent transactions.
+
+| Name | SSZ equivalent |
+| - | - |
+| [`BLSFieldElement`](./eip-4844.md#type-aliases) | `uint256` |
+| [`VersionedHash`](./eip-4844.md#type-aliases) | `Bytes32` |
+| [`KZGCommitment`](./eip-4844.md#type-aliases) | `Bytes48` |
+| [`KZGProof`](./eip-4844.md#type-aliases) | `Bytes48` |
+
+| Name | Value |
+| - | - |
+| [`FIELD_ELEMENTS_PER_BLOB`](./eip-4844.md#parameters) | `uint64(2**12)` (= 4,096) |
+| [`MAX_VERSIONED_HASHES_LIST_SIZE`](./eip-4844.md#parameters) | `uint64(2**24)` (= 16,777,216) |
+| [`MAX_CALLDATA_SIZE`](./eip-4844.md#parameters) | `uint64(2**24)` (= 16,777,216) |
+| [`MAX_ACCESS_LIST_SIZE`](./eip-4844.md#parameters) | `uint64(2**24)` (= 16,777,216) |
+| [`MAX_ACCESS_LIST_STORAGE_KEYS`](./eip-4844.md#parameters) | `uint64(2**24)` (= 16,777,216) |
+| [`MAX_TX_WRAP_KZG_COMMITMENTS`](./eip-4844.md#parameters) | `uint64(2**24)` (= 16,777,216) |
+| [`LIMIT_BLOBS_PER_TX`](./eip-4844.md#parameters) | `uint64(2**24)` (= 16,777,216) |
+
+```python
+class AccessTuple(Container):
+    address: Address
+    storage_keys: List[Hash, MAX_ACCESS_LIST_STORAGE_KEYS]
+
+class Transaction(Container):
+    chain_id: uint64  # EIP-155
+    nonce: uint64
+    max_priority_fee_per_gas: uint256  # EIP-1559
+    max_fee_per_gas: uint256  # aka `gasprice`
+    gas_limit: uint64  # aka `startgas`
+    to: Optional[Address]  # None: deploy contract
+    value: uint256
+    data: ByteList[MAX_CALLDATA_SIZE]
+    access_list: List[AccessTuple, MAX_ACCESS_LIST_SIZE]  # EIP-2930
+    blob_versioned_hashes: List[VersionedHash, MAX_VERSIONED_HASHES_LIST_SIZE]  # EIP-4844
+
+class ECDSASignature(Container):
+    y_parity: boolean  # EIP-2930
+    r: uint256
+    s: uint256
+
+class SignedTransaction(Container):
+    tx: Transaction
+    signature: ECDSASignature
+    hash_version: TxHashVersion
+
+class BlobTransactionExtension(Container):
+    blob_kzgs: List[KZGCommitment, MAX_TX_WRAP_KZG_COMMITMENTS]
+    blobs: List[Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB], LIMIT_BLOBS_PER_TX]
+    kzg_aggregated_proof: KZGProof
+
+class TransactionPayload(Container):
+    signed_tx: SignedTransaction
+    blob: Optional[BlobTransactionExtension]  # EIP-4844
+```
+
+On the network, `TransactionPayload` is preceded by a single byte, [`BLOB_TX_TYPE`](./eip-4844.md#parameters) to distinguish from other transaction formats according to [EIP-2718](./eip-2718.md), regardless of the inner `signed_tx.hash_version`.
+
+The `blocks.transactions` list remains RLP encoded:
```

Comment:
Where is `blocks` defined? Also, I don't think this is necessary. How the list of transactions is encoded is not in consensus, only defined in the devp2p spec.

Reply:
That's right, I'm referring to that `block` from devp2p.
https://github.com/ethereum/devp2p/blob/master/caps/eth.md#block-encoding-and-validity

And agree, the `transactions` are already specified to pick up the `TransactionPayload` format based on the EIP-2718 prefix. So, mentioning it here is indeed not necessary.